src/bin/console/console.cpp

```diff
@@ -718,40 +719,30 @@ int Console::_visualize(const std::string& input) {
     case PlanType::UnoptLQP: {
       std::vector<std::shared_ptr<AbstractLQPNode>> lqp_roots;
 
-      try {
-        const auto& lqps = (plan_type == PlanType::LQP) ? _sql_pipeline->get_optimized_logical_plans()
-                                                        : _sql_pipeline->get_unoptimized_logical_plans();
-        for (const auto& lqp : lqps) {
-          lqp_roots.push_back(lqp);
-        }
-      } catch (const std::exception& exception) {
-        out(std::string(exception.what()) + "\n");
-        return ReturnCode::Error;
+      const auto& lqps = (plan_type == PlanType::LQP) ? _sql_pipeline->get_optimized_logical_plans()
+                                                      : _sql_pipeline->get_unoptimized_logical_plans();
+
+      lqp_roots.reserve(lqps.size());
+
+      for (const auto& lqp : lqps) {
+        lqp_roots.push_back(lqp);
```

Comment:
Let's use `emplace_back` to avoid the extra copy or move operation

Reply:
Out of curiosity: does that change anything? We need to add a new copy of the inital lqp shrptr anyways and the loop just holds a reference. So we cannot move it.

Or is there more behind it?

Reply:
Probably not.

https://stackoverflow.com/questions/10890653/why-would-i-ever-use-push-back-instead-of-emplace-back

Reply:
Both `push_back` and `emplace_back` call the constructor but only `emplace_back` construct the element in-place at the location provided by the container. `push_back` creates a temporary object which is then moved to the new position in the container.

Please correct me if I'm wrong.

Reply:
https://wandbox.org/permlink/Q9TV2fIzhRASFOHv